Fail to install Tye CI build 0.11.0

**OS:** Windows 10

**Repro Steps:**
1. Install .NET Core 3.1.
2. Install  Tye CI build using [the command](https://github.com/dotnet/tye/blob/main/docs/getting_started.md#working-with-ci-builds).
`dotnet tool install -g Microsoft.Tye --version "0.11.0-*" --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json`

**Expect:**
Succeed to install Tye CI build 0.11.0.

**Actual:**
Fail to install Tye CI build 0.11.0.
![image](https://user-images.githubusercontent.com/45614674/136640432-f43b5b99-a1e3-4392-a466-45f163c58a53.png)

**Error Info:**
```
C:\WINDOWS\system32>dotnet tool install -g Microsoft.Tye --version "0.11.0-*" --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json

Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.413

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------
error NU1102: Unable to find package microsoft.tye with version (>= 0.11.0)
error NU1102:   - Found 328 version(s) in https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json [ Nearest version: 0.10.0-alpha.21420.1 ]
error NU1102:   - Found 11 version(s) in nuget.org [ Nearest version: 0.10.0-alpha.21420.1 ]
The tool package could not be restored.
Tool 'microsoft.tye' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET Core tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

```

--- a/eng/Signing.props
+++ b/eng/Signing.props
@@ -14,6 +14,8 @@
     <FileSignInfo Include="OpenTelemetry.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="OpenTelemetry.Api.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="OpenTelemetry.Exporter.Zipkin.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Prometheus.NetCore.dll" CertificateName="3PartySHA2" />
+    <FileSignInfo Include="Prometheus.NetStandard.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="Semver.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="Serilog.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="Serilog.Extensions.Hosting.dll" CertificateName="3PartySHA2" />
@@ -27,6 +29,7 @@
     <FileSignInfo Include="Serilog.Sinks.Seq.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="Sln.dll" CertificateName="3PartySHA2" />
     <!-- Despite the System. nomenclature, this is a non-MSFT .NET foundation project.-->
+    <FileSignInfo Include="System.IO.Abstractions.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="System.Reactive.dll" CertificateName="3PartySHA2" />
     <FileSignInfo Include="YamlDotNet.dll" CertificateName="3PartySHA2" />
   </ItemGroup>

--- a/src/Microsoft.Tye.Core/Microsoft.Tye.Core.csproj
+++ b/src/Microsoft.Tye.Core/Microsoft.Tye.Core.csproj
@@ -8,17 +8,17 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="KubernetesClient" Version="4.0.10" />
+    <PackageReference Include="KubernetesClient" Version="6.0.25" />
     
     <PackageReference Include="Microsoft.Win32.Registry" Version="4.7.0" />
     <PackageReference Include="semver" Version="2.0.6" />
     <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20071.2" />
     <PackageReference Include="System.Commandline.Rendering" Version="0.3.0-alpha.20070.2" />
-    <PackageReference Include="System.Security.Cryptography.Cng" Version="4.7.0" />
+    <PackageReference Include="System.Security.Cryptography.Cng" Version="5.0.0" />
     <!-- Hoisted to avoid a conflict with Microsoft.Build -->
-    <PackageReference Include="System.Security.Principal.Windows" Version="4.7.0" />
-    <PackageReference Include="YamlDotNet" Version="8.1.2" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.5" />
+    <PackageReference Include="System.Security.Principal.Windows" Version="5.0.0" />
+    <PackageReference Include="YamlDotNet" Version="11.2.1" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Microsoft.Tye.Hosting.Diagnostics/Logging/LogObject.cs
+++ b/src/Microsoft.Tye.Hosting.Diagnostics/Logging/LogObject.cs
@@ -11,7 +11,7 @@ namespace Microsoft.Tye.Hosting.Diagnostics.Logging
 {
     internal class LogObject : IReadOnlyList<KeyValuePair<string, object?>>
     {
-        internal static readonly Func<object, Exception, string> Callback = (state, exception) => ((LogObject)state).ToString();
+        internal static readonly Func<object, Exception?, string> Callback = (state, exception) => ((LogObject)state).ToString();
 
         private readonly string? _formattedMessage;
         private List<KeyValuePair<string, object?>> _items = new List<KeyValuePair<string, object?>>();

--- a/src/Microsoft.Tye.Hosting.Diagnostics/Microsoft.Tye.Hosting.Diagnostics.csproj
+++ b/src/Microsoft.Tye.Hosting.Diagnostics/Microsoft.Tye.Hosting.Diagnostics.csproj
@@ -25,8 +25,8 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent" Version="2.0.49" GeneratePathProperty="true" />
     <PackageReference Include="Microsoft.Diagnostics.NETCore.Client" Version="0.2.61701" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.1" />
-    <PackageReference Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.12.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.19.0" />
     <PackageReference Include="OpenTelemetry" Version="0.2.0-alpha.179" />
     <PackageReference Include="OpenTelemetry.Exporter.Zipkin" Version="0.2.0-alpha.179" />
     <PackageReference Include="Serilog.Extensions.Logging" Version="3.0.0" />

--- a/src/Microsoft.Tye.Hosting/Microsoft.Tye.Hosting.csproj
+++ b/src/Microsoft.Tye.Hosting/Microsoft.Tye.Hosting.csproj
@@ -25,7 +25,7 @@
     <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
     <PackageReference Include="System.IO.Pipelines" Version="4.7.2" />
     <PackageReference Include="System.Reactive" Version="4.3.2" />
-    <PackageReference Include="YamlDotNet" Version="8.1.2" />
+    <PackageReference Include="YamlDotNet" Version="11.2.1" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/tye/tye.csproj
+++ b/src/tye/tye.csproj
@@ -27,7 +27,7 @@
 
   <ItemGroup>
     <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20104.2" />
-    <PackageReference Include="YamlDotNet" Version="8.1.2" />
+    <PackageReference Include="YamlDotNet" Version="11.2.1" />
   </ItemGroup>
 
 </Project>

--- a/test/Test.Infrastructure/EqualityYamlNodeVisitor.cs
+++ b/test/Test.Infrastructure/EqualityYamlNodeVisitor.cs
@@ -39,7 +39,7 @@ namespace Test.Infrastructure
                     $"Actual: {node.NodeType} at ({otherNode.Start.Line}, {otherNode.Start.Column}).");
             }
 
-            if (node.Tag?.Equals(otherNode.Tag) == false)
+            if (node.Tag.Equals(otherNode.Tag) == false)
             {
                 throw new TyeYamlException($"Expected tags to be equal. " +
                     $"Expected: ({node.Start.Line}, {node.Start.Column}). Actual: ({otherNode.Start.Line}, {otherNode.Start.Column}).");

